pkg/virt-launcher/virtwrap/network/podnic_test.go

```diff
@@ -198,6 +197,116 @@ var _ = Describe("podNIC", func() {
 			})
 		})
 	})
+	type stateRetrievalTestCase struct {
+		storedState   *cache.PodIfaceState
+		expectedState cache.PodIfaceState
+	}
+	statePointer := func(state cache.PodIfaceState) *cache.PodIfaceState {
+		return &state
+	}
+	DescribeTable("state is retrieved", func(c stateRetrievalTestCase) {
```

Comment:
I'm unsure how I feel about specifying the behavior of the code by its output (i.e. passing expected state as input to the describe table's function).

Can't you add a different clause for that scenario ? I think it would be natural, since - to me - it is pretty obvious you're using two different `When` clauses - "when there is no pod interface cache" and "when the caches are provisioned".